lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +191,144 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Trims the databases to the block prior to the block the given transaction is in.
-   * @returns The last fully processed block after trimming. `undefined` if all data are deleted after trimming.
+   * A faster version of process transactions that requires access to bitcoin data directory
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const bitcoinBlockDataIterator = new BitcoinBlockDataIterator(this.bitcoinDataDirectory!);
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let currentHeight = bestHeight;
+    const numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    // An array of maps<hash, IBlockInfoExtended>.
+    const chainInfos = [...new Array(numOfBlocksToProcess)].map<Map<string, IBlockInfoExtended>>(() => new Map());
+
+    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
+    // loop through files and process them until we process all blocks needed
+    while (bitcoinBlockDataIterator.hasNext() && currentHeight >= startingBlock.height) {
+      const blockData = bitcoinBlockDataIterator.next()!;
+      await this.processBlockData(blockData, chainInfos, startingBlock.height, currentHeight);
```

Comment:
I start to have issue with the name `BlockData` throughout the code as I have on multiple occasions thinking it is data on a single block, but only to realize that it is multiple of them, would love to have a better name.

Reply:
just `blocks`?

Reply:
Sounds good, but the monkey wrench is `BitcoinBlockDataIterator` would become `BitcoinBlocksIterator`.... Maybe for that class we call it.... `BitcoinBlockFileIterator`?

Reply:
I think BlockDataIterator still makes sense because we are iterating through the raw block data, processing them and then return the processed result. Will change it if you really don't like it thought.